flink-table/flink-table-planner-blink/src/test/java/org/apache/flink/table/planner/expressions/CompositeTypeAccessExpressionITCase.java

```diff
@@ -0,0 +1,264 @@
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.planner.expressions;
+
+import org.apache.flink.table.annotation.DataTypeHint;
+import org.apache.flink.table.annotation.FunctionHint;
+import org.apache.flink.table.api.DataTypes;
+import org.apache.flink.table.api.EnvironmentSettings;
+import org.apache.flink.table.api.TableEnvironment;
+import org.apache.flink.table.api.TableResult;
+import org.apache.flink.table.api.TableSchema;
+import org.apache.flink.table.api.ValidationException;
+import org.apache.flink.table.functions.BuiltInFunctionDefinitions;
+import org.apache.flink.table.functions.ScalarFunction;
+import org.apache.flink.table.functions.TableFunction;
+import org.apache.flink.table.types.logical.LogicalTypeRoot;
+import org.apache.flink.types.Row;
+import org.apache.flink.util.CloseableIterator;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Suite;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static java.util.Collections.singletonMap;
+import static org.apache.flink.table.api.DataTypes.ARRAY;
+import static org.apache.flink.table.api.DataTypes.BIGINT;
+import static org.apache.flink.table.api.DataTypes.FIELD;
+import static org.apache.flink.table.api.DataTypes.MAP;
+import static org.apache.flink.table.api.DataTypes.ROW;
+import static org.apache.flink.table.api.DataTypes.STRING;
+import static org.apache.flink.table.api.Expressions.$;
+import static org.apache.flink.table.api.Expressions.call;
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertThat;
+
+/**
+ * Tests for functions that access nested fields/elements of composite/collection types.
+ */
+@RunWith(Suite.class)
+@Suite.SuiteClasses(
+	{
+		CompositeTypeAccessExpressionITCase.TableFieldAccess.class,
+		CompositeTypeAccessExpressionITCase.CallFieldAccess.class
+	}
+)
+public class CompositeTypeAccessExpressionITCase {
+
+	/**
+	 * Regular tests. See also {@link CallFieldAccess} for tests that access a nested field of an expression or
+	 * for {@link BuiltInFunctionDefinitions#FLATTEN} which produces multiple columns from a single one.
+	 */
+	public static class TableFieldAccess extends BuiltInFunctionTestBase {
+		@Parameterized.Parameters(name = "{index}: {0}")
+		public static List<TestSpec> testData() {
+			return Arrays.asList(
+
+				// Actually in case of SQL it does not use the GET method, but
+				// a custom logic for accessing nested fields of a Table.
+				TestSpec.forFunction(BuiltInFunctionDefinitions.GET)
+					.onFieldsWithData(null, Row.of(1))
+					.andDataTypes(
+						ROW(FIELD("nested", BIGINT().notNull())).nullable(),
+						ROW(FIELD("nested", BIGINT().notNull())).notNull()
+					)
+					.testTableApiResult($("f0").get("nested"), null, BIGINT().nullable())
```

Comment:
use the newly introduced `testResult` such that a result must be defined only once